cmd/gardener-extension-shoot-dns-service/app/app.go

```diff
@@ -61,19 +65,28 @@ func (o *Options) run(ctx context.Context) {
 		Burst: 130,
 	}, o.restOptions.Completed().Config)
 
-	mgr, err := manager.New(o.restOptions.Completed().Config, o.managerOptions.Completed().Options())
-	if err != nil {
-		controllercmd.LogErrAndExit(err, "Could not instantiate controller-manager")
-	}
+	mgrOpts := o.managerOptions.Completed().Options()
 
-	if err := dnsapi.AddToScheme(mgr.GetScheme()); err != nil {
+	if mgrOpts.Scheme == nil {
+		mgrOpts.Scheme = scheme.Scheme
+	}
+	if err := dnsapi.AddToScheme(mgrOpts.Scheme); err != nil {
 		controllercmd.LogErrAndExit(err, "Could not update manager scheme (dnsapi)")
 	}
-
-	if err := extensionscontroller.AddToScheme(mgr.GetScheme()); err != nil {
+	if err := extensionscontroller.AddToScheme(mgrOpts.Scheme); err != nil {
 		controllercmd.LogErrAndExit(err, "Could not update manager scheme")
 	}
 
+	mgrOpts.NewClient = utils.NewClientFuncWithDisabledCacheFor(
+		&corev1.Secret{},    // applied for ManagedResources
+		&corev1.ConfigMap{}, // applied for monitoring config
+		&dnsapi.DNSOwner{},  // avoid watching DNSOwner
```

Comment:
Are the `DNSOwner` resources updated very frequently or are they very huge?
When does this extension interact with them?

Reply:
A single DNSOwner resource is accessed on startup. If it would not be excluded, all DNSOwner resources would be cached and watched although they are never needed again.

Reply:
Ok, then it makes sense to exclude them. Thanks for the explanation.